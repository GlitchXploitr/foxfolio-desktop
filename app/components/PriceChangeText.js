// @flow
import React from 'react';
import { withStyles } from 'material-ui';

type Props = {
  change: number,
  classes: Object
};

const styles = theme => ({
  negative: {
    color: theme.palette.error[500],
  },
  positive: {
    color: theme.palette.success[500],
  },
});

function PriceChangeText({ change, classes }: Props) {
  return (
    <span className={change > 0 ? classes.positive : classes.negative}>
      {`${change > 0 ? '+' : ''}${(change).toFixed(2)}%`}
    </span>
  );
}

export default withStyles(styles)(PriceChangeText);
