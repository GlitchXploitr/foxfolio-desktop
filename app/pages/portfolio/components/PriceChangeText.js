// @flow
import React from 'react';
import { withStyles } from 'material-ui';
import { green, grey, red } from 'material-ui/colors';

type Props = {
  change: number,
  muted: boolean,
  classes: Object
};

const styles = () => ({
  negative: {
    color: red[500],
  },
  negativeMuted: {
    color: red[300],
  },
  positive: {
    color: green[500],
  },
  positiveMuted: {
    color: green[300],
  },
  interval: {
    fontSize: 12,
    color: grey[700],
  },
  intervalMuted: {
    fontSize: 12,
    color: grey[500],
  },
});

function PriceChangeText({ change, classes, muted }: Props) {
  let className = classes.positive;
  if (change > 0 && muted) {
    className = classes.positiveMuted;
  } else if (change < 0 && !muted) {
    className = classes.negative;
  } else if (change < 0 && muted) {
    className = classes.negativeMuted;
  }
  return (
    <div style={{ display: 'inline-block' }}>
      <span className={className}>
        {`${change > 0 ? '+' : ''}${(change).toFixed(2)}%`}
      </span>
      <span className={muted ? classes.intervalMuted : classes.interval}> /24h</span>
    </div>
  );
}

export default withStyles(styles)(PriceChangeText);
