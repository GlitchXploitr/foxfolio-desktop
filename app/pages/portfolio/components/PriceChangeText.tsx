// @flow
import { withStyles } from 'material-ui';
import { green, grey, red } from 'material-ui/colors';
import React from 'react';

interface Props {
  change: number;
  muted?: boolean;
}

const styles = () => ({
  negative: {
    color: red[500],
  },
  negativeMuted: {
    color: red[300],
  },
  positive: {
    color: green[500],
  },
  positiveMuted: {
    color: green[300],
  },
  interval: {
    fontSize: 12,
    color: grey[700],
  },
  intervalMuted: {
    fontSize: 12,
    color: grey[500],
  },
});

export const PriceChangeText = withStyles(styles)<Props>(({ change, classes, muted }) => {
  let className = classes.positive;
  if (change > 0 && muted) {
    className = classes.positiveMuted;
  } else if (change < 0 && !muted) {
    className = classes.negative;
  } else if (change < 0 && muted) {
    className = classes.negativeMuted;
  }
  return (
    <div style={{ display: 'inline-block' }}>
      <span className={className}>{`${change > 0 ? '+' : ''}${change.toFixed(2)}%`}</span>
      <span className={muted ? classes.intervalMuted : classes.interval}> /24h</span>
    </div>
  );
});
