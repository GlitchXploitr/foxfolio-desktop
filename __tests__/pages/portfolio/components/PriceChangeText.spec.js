// @flow
import * as React from 'react';
import { createShallow } from 'material-ui/test-utils';
import PriceChangeText from '../../../../app/pages/portfolio/components/PriceChangeText';

let shallow;

beforeAll(() => {
  shallow = createShallow();
});

test('Positive change', () => {
  const wrapper = shallow(<PriceChangeText change={5}/>).dive();
  expect(wrapper).toMatchSnapshot();
});

test('Negative change', () => {
  const wrapper = shallow(<PriceChangeText change={-5}/>).dive();
  expect(wrapper).toMatchSnapshot();
});
