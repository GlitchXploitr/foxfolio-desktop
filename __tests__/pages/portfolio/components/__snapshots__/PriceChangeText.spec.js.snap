// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Negative change 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <span
    className="PriceChangeText-negative-1"
  >
    -5.00%
  </span>
  <span
    className="PriceChangeText-interval-5"
  >
     /24h
  </span>
</div>
`;

exports[`Positive change 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <span
    className="PriceChangeText-positive-3"
  >
    +5.00%
  </span>
  <span
    className="PriceChangeText-interval-5"
  >
     /24h
  </span>
</div>
`;
